### PR TITLE
fix: update team id prop for hosts and visitors

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at c04e9cc</samp>

This pull request adds team support to the visitors report and the host creation features in the journey editor. It uses the `useTeam` hook, the `TeamProvider` component, and the `GET_TEAMS` query to fetch and provide the active team for the current user, and filters the visitors and hosts accordingly. It also updates the tests to reflect the team-based logic.

- [Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/33034379/todos/6371780677)
- [Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/33034379/todos/6371776027)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it should accept teamID prop in host create and get visitors calls

# Walkthrough

### <samp>🤖 Generated by Copilot at c04e9cc</samp>

*  Use the `useTeam` hook to fetch and provide the active team information for the current user in various components and queries ([link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8R30), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8R85), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L118-R121),  [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-c7e0b31cbea63567e14f5550d0735a8acb8fae18f0282f55cd021845d9f73ff7L110-R137), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-11b3f1d4f5e0d25d0b497fdddc12a9f439c7a2c59c30647b9f01304c72b9e93fR9), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-11b3f1d4f5e0d25d0b497fdddc12a9f439c7a2c59c30647b9f01304c72b9e93fL34-R36), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-685fc29c8e3bfe137947aaca1a22de99e237dfb9a33bd37a5693cd5c5d402682R14), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-685fc29c8e3bfe137947aaca1a22de99e237dfb9a33bd37a5693cd5c5d402682R67-R68), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-685fc29c8e3bfe137947aaca1a22de99e237dfb9a33bd37a5693cd5c5d402682L74-R78))
* Replace the hard-coded teamId values with variables that match the active team id or the mock results for the `GET_TEAMS` query in `visitors.tsx`, `HostTitleFieldForm.spec.tsx`, and `HostTitleFieldForm.tsx` ([link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L39-R40), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-c7e0b31cbea63567e14f5550d0735a8acb8fae18f0282f55cd021845d9f73ff7L91-R101), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-11b3f1d4f5e0d25d0b497fdddc12a9f439c7a2c59c30647b9f01304c72b9e93fL47-R51))
* Add the `teamId` variable to the mock requests and results for the `GET_VISITORS` query in `VisitorList.spec.tsx` ([link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fL27-R30), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fL146-R174))
* Create mock results for the `GET_TEAMS` query and wrap the components with the `TeamProvider` component in `HostTitleFieldForm.spec.tsx` and `VisitorList.spec.tsx` ([link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-c7e0b31cbea63567e14f5550d0735a8acb8fae18f0282f55cd021845d9f73ff7R87-R92), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fR63-R73), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fL68-R88), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fL168-R192), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fL192-R218), [link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-72cd73ec3f1b6d5c4a1dca55fefd1069aefb1965888af95793b45511d222c33fL210-R238))
* Add a condition to check if the active team is not null before calling the `hostCreate` mutation in `HostTitleFieldForm.tsx` ([link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-11b3f1d4f5e0d25d0b497fdddc12a9f439c7a2c59c30647b9f01304c72b9e93fL47-R51))
* Add a new test case to check that the host creation does not happen if the active team id is null in `HostTitleFieldForm.spec.tsx` ([link](https://github.com/JesusFilm/core/pull/1722/files?diff=unified&w=0#diff-c7e0b31cbea63567e14f5550d0735a8acb8fae18f0282f55cd021845d9f73ff7R161-R242))

### CODE REVIEW

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx [hunk 1]
```diff
@@ -115,7 +117,8 @@ function JourneyVisitorsPage(): ReactElement {
           hideInactive: hideInteractive
         },
         first: 100,
-        sort: sortSetting
+        sort: sortSetting,
+        teamId: activeTeam?.id
```
Comment (tataihono) [inline apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:121]: what happens if this is null? I think we need to enforce this.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/VisitorsList/VisitorsList.tsx [hunk 2]
```diff
@@ -71,7 +73,8 @@ export function VisitorsList(): ReactElement {
 
   const { fetchMore, loading } = useQuery<GetVisitors>(GET_VISITORS, {
     variables: {
-      first: 100
+      first: 100,
+      teamId: activeTeam?.id
```
Comment (storyworks) [inline apps/journeys-admin/src/components/VisitorsList/VisitorsList.tsx:77]: Enforce this too

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Footer/HostSidePanel/HostForm/HostTitleFieldForm/HostTitleFieldForm.tsx [hunk 3]
```diff
@@ -44,9 +45,9 @@ export function HostTitleFieldForm(): ReactElement {
     if (host != null) {
       const { id, teamId } = host
       await updateHost({ id, teamId, input: { title: value } })
-    } else {
+    } else if (activeTeam != null) {
       const { data } = await hostCreate({
-        variables: { teamId: 'jfp-team', input: { title: value } },
+        variables: { teamId: activeTeam?.id, input: { title: value } },
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Footer/HostSidePanel/HostForm/HostTitleFieldForm/HostTitleFieldForm.tsx:50]: Don't need the `?` here since activeTeam will exist

**Review by storyworks — APPROVED** [review]

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx [hunk 4]
```diff
@@ -126,12 +129,13 @@ function JourneyVisitorsPage(): ReactElement {
...
   async function handleFetchNext(): Promise<void> {
-    if (visitorEdges != null && hasNextPage) {
+    if (visitorEdges != null && hasNextPage && activeTeam != null) {
       const response = await fetchMore({
         variables: {
           filter: { journeyId },
           first: 100,
-          after: endCursor
+          after: endCursor,
+          teamId: activeTeam.id
```
Comment (tataihono) [inline apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:138]: drop

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx [hunk 5]
```diff
@@ -27,16 +27,18 @@ import { VisitorToolbar } from '../../../../src/components/JourneyVisitorsList/V
...
   query GetJourneyVisitors(
     $filter: JourneyVisitorFilter!
     $sort: JourneyVisitorSort
     $first: Int
     $after: String
+    $teamId: String!
   ) {
     visitors: journeyVisitorsConnection(
-      teamId: "jfp-team"
+      teamId: $teamId
```
Comment (tataihono) [inline apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:41]: drop

File: apps/journeys-admin/src/components/VisitorsList/VisitorsList.tsx [hunk 6]
```diff
@@ -11,13 +11,14 @@ import {
...
+import { useTeam } from '../Team/TeamProvider'
 import { getColDefs } from './utils/getColDefs'
 import { getVisitorRows } from './utils/getVisitorRows'
 
 export const GET_VISITORS = gql`
-  query GetVisitors($first: Int, $after: String) {
+  query GetVisitors($first: Int, $after: String, $teamId: String!) {
     visitors: visitorsConnection(
-      teamId: "jfp-team"
+      teamId: $teamId
```
Comment (tataihono) [inline apps/journeys-admin/src/components/VisitorsList/VisitorsList.tsx:21]: drop

23-05-TN-fix-visitor-apis-teams

File: apps/journeys-admin/src/components/VisitorsList/VisitorsList.tsx [hunk 7]
```diff
@@ -81,11 +83,12 @@ export function VisitorsList(): ReactElement {
...
 
   async function handleFetchNext(): Promise<void> {
-    if (hasNextPage) {
+    if (hasNextPage && activeTeam != null) {
       const response = await fetchMore({
         variables: {
           first: 100,
-          after: endCursor
+          after: endCursor,
+          teamId: activeTeam.id
```
Comment (tataihono) [inline apps/journeys-admin/src/components/VisitorsList/VisitorsList.tsx:91]: drop

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Review by tataihono — APPROVED** [review]
